pkg/controller/controller_utils.go

```diff
@@ -517,29 +519,67 @@ func validateControllerRef(controllerRef *metav1.OwnerReference) error {
 }
 
 func (r RealPodControl) CreatePods(namespace string, template *v1.PodTemplateSpec, object runtime.Object) error {
-	return r.createPods("", namespace, template, object, nil)
+	pod, err := GetPodFromTemplate(template, object)
+	if err != nil {
+		return err
+	}
+	return r.createPods(namespace, pod, object)
 }
 
 func (r RealPodControl) CreatePodsWithControllerRef(namespace string, template *v1.PodTemplateSpec, controllerObject runtime.Object, controllerRef *metav1.OwnerReference) error {
 	if err := validateControllerRef(controllerRef); err != nil {
 		return err
 	}
-	return r.createPods("", namespace, template, controllerObject, controllerRef)
+	pod, err := GetPodFromTemplate(template, controllerObject)
+	if err != nil {
+		return err
+	}
+	if controllerRef != nil {
+		pod.OwnerReferences = append(pod.OwnerReferences, *controllerRef)
+	}
+	return r.createPods(namespace, pod, controllerObject)
 }
 
 func (r RealPodControl) CreatePodsOnNode(nodeName, namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference) error {
 	if err := validateControllerRef(controllerRef); err != nil {
 		return err
 	}
-	return r.createPods(nodeName, namespace, template, object, controllerRef)
+	pod, err := GetPodFromTemplate(template, object)
```

Comment:
just to avoid this change, hopefully we can merge #101156 first

@soltysh can you approve? :)